lib/util/git/private-key.spec.ts

```diff
@@ -10,39 +13,119 @@ jest.mock('fs-extra', () =>
     >('../../../test/fixtures')
     .fsExtra(),
 );
-jest.mock('../exec');
+jest.mock('../exec', () => mockDeep());
 
-const exec = mocked(exec_);
+const exec = mockedExtended(exec_);
 
 describe('util/git/private-key', () => {
   describe('writePrivateKey()', () => {
+    beforeEach(() => {
+      Fixtures.reset();
+    });
+
     it('returns if no private key', async () => {
       await expect(writePrivateKey()).resolves.not.toThrow();
       await expect(configSigningKey('/tmp/some-repo')).resolves.not.toThrow();
     });
 
     it('throws error if failing', async () => {
       setPrivateKey('some-key');
-      exec.exec.mockRejectedValueOnce({
-        stderr: `something wrong`,
-        stdout: '',
-      });
+      exec.exec.calledWith(any()).mockResolvedValue({ stdout: '', stderr: '' });
+      exec.exec
+        .calledWith('gpg --import /tmp/git-private-gpg.key')
+        .mockRejectedValueOnce({
+          stderr: `something wrong`,
+          stdout: '',
+        });
```

Comment:
please do these refactorings in a separate PR. Ideallay this PR doesn't change any existing test to ensure no regressions

Reply:
Done, see #30396.